FEATURE: add configuration option to use custom form factories

andrehoffmann30 opened the pull request:
Adds an option to configure a custom form factory. This factory is then used for generating the form definition object which defines the form elements to be rendered. The rendering of the form is handle by the render viewhelper from the neos/form package respectively by the renderformpageAction. The use of the custom form factory allows the developer to add additional data to the form elements e.g. from the data base. In my case, I needed the option to add data about car makers from the project data base to a carmaker form element.

andrehoffmann30 committed:
FEATURE: add configuration option to use custom form factories

andrehoffmann30 committed:
TASK: allow use of neos/form ^5.0

andrehoffmann30 committed:
TASK: adjust json configration viewhelper

bwaidelich commented:
Hi @andrehoffmann30 

Thanks a lot for your contribution!
Could you please add a short description to this PR that explains what this is about and shows a simple usage example?

andrehoffmann30 committed:
TASK: adjust readme

andrehoffmann30 commented:
Hi @bwaidelich 
I added an option to configure a custom form factory. This factory is then used for generating the form definition object which  defines the form elements to be rendered. The rendering of the form is handle by the render viewhelper from the neos/form package respectively by the renderformpageAction. The use of the custom form factory allows the developer to add additional data to the form elements e.g. from the data base. In my case, I needed the option to add data about car makers from the project data base to a carmaker form element. I have added a configuration example to the readme and also to the settings.yaml, as a comment. Can you please let me know if you need me to do anything further explanation. Thank you. 
Can you also review the code and let me now if some adjustments are need. 😄
